tests/gene-comparison-tool-pinning-url.spec.ts

```diff
@@ -0,0 +1,337 @@
+  });
+
+  test('when RNA url includes 50 pinned genes, all genes are pinned', async ({
+    page,
+  }) => {
+    const url = `${URL_GCT}?${formatPinnedGenesQueryParam(fiftyGenes)}`;
+    await page.goto(url);
+
+    await expectGctPageLoaded(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD
+    );
+
+    await test.step('confirm pinned all 50 genes', async () => {
+      await expect(page.getByText('All Genes')).toBeVisible();
+      await expectPinnedGenesCountText(page, 50);
+      await confirmPinnedItemsCount(page, 50);
+      await expect(page.getByRole('alert')).not.toBeVisible();
+    });
+
+    await test.step('confirm cannot pin other genes', async () => {
+      const btns = await getGeneRowButtons(page, 'PYGL');
+      await expect(btns.open).not.toBeDisabled();
+      await expect(btns.pin).toBeDisabled();
+    });
+  });
+
+  test('when RNA url includes >50 pinned genes, only 50 genes are pinned and toast is displayed', async ({
+    page,
+  }) => {
+    const url = `${URL_GCT}?${formatPinnedGenesQueryParam(fiftyOneGenes)}`;
+    await page.goto(url);
+
+    await expectGctPageLoaded(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD
+    );
+
+    await test.step('confirm only pinned 50 genes', async () => {
+      await expect(page.getByText('All Genes')).toBeVisible();
+      await expectPinnedGenesCountText(page, 50);
+      await confirmPinnedItemsCount(page, 50);
+      await expectTooManyPinnedGenesToast(page);
+    });
+
+    await test.step('confirm url dropped 51st gene', () => {
+      const genes = getPinnedItemsFromUrl(page.url());
+      expect(genes).toHaveLength(50);
+      expect(genes).toEqual(fiftyOneGenes.sort().slice(0, -1));
+    });
+  });
+
+  test('when RNA url includes invalid gene, that gene is dropped from the url', async ({
+    page,
+  }) => {
+    const validGeneId = geneWithMultipleProteinsTMT.ensemblId;
+    const url = `${URL_GCT}?${formatPinnedGenesQueryParam([
+      validGeneId,
+      'invalidGene',
+    ])}`;
+    await page.goto(url);
+
+    await expectGctPageLoaded(
+      page,
+      GCT_CATEGORIES.RNA,
+      GCT_RNA_SUBCATEGORIES.AD
+    );
+
+    await test.step('confirm toast not shown', async () => {
+      await expect(page.getByRole('alert')).not.toBeVisible();
+    });
+
+    await test.step('confirm only pinned 1 gene', async () => {
+      await expect(page.getByText('All Genes')).toBeVisible();
+      await expectPinnedGenesCountText(page, 1);
+      await confirmPinnedItemsCount(page, 1);
+    });
+
+    await test.step('confirm url dropped invalid gene', () => {
+      const genes = getPinnedItemsFromUrl(page.url());
+      expect(genes).toHaveLength(1);
+      expect(genes).toEqual([validGeneId]);
+    });
+  });
+
+  test.fail('when RNA url includes proteins, the related gene is pinned', {
+    annotation: {
+      type: 'fail', 
+      description: 'Since AG-1425, only genes will be pinned from RNA url'
+    }
+  },
```

Comment:
Expect this test to fail -- since AG-1425, only genes will be pinned from an RNA url. ensembl id + uniprot id will be dropped.